civihr_employee_portal/civihr_employee_portal.module

```diff
@@ -777,6 +778,45 @@ function get_civihr_absence_statuses($status_id = NULL) {
     return $status_id_value;
 }
 
+/**
+ * This function returns the sickness reason label for the sickness reason
+ * option value passed to it.
+ * The sickness reason option group is cached to avoid hitting the Db on
+ * every call.
+ *
+ * @param int $sickness_reason_value
+ *
+ * @return string
+ */
+function get_civihr_absence_sickness_reason_label($sickness_reason_value) {
+    $sicknessReasons = &drupal_static(__FUNCTION__);
+
+    if (!isset($sicknessReasons)) {
+        if ($cache = cache_get('civihr_absence_sickness_reasons')) {
+            $sicknessReasons = $cache->data;
+        }
+        else {
+            try {
+                // Civi init
+                civicrm_initialize();
+                $sicknessReasons = CRM_Core_OptionGroup::values('hrleaveandabsences_sickness_reason', FALSE, FALSE, FALSE, NULL, 'label');
+
+                watchdog('DB hit absence sickness reasons', 'DB');
+            }
+
+            catch (CiviCRM_API3_Exception $e) {
+                $error = $e->getMessage();
+            }
+
+            // Cache the absence sickness reasons for 5 minutes
+            cache_set('civihr_absence_sickness_reasons', $sicknessReasons, 'cache', time() + 360);
+        }
+    }
+    $sickness_reason_label = CRM_Utils_Array::value($sickness_reason_value, $sicknessReasons);
+
+    return $sickness_reason_label;
+}
```

Comment:
so basically, I would replace this function with something like this : 

```php
function get_civihr_absence_sickness_reason_label($sicknessReasonValue) {
  $sicknessReasons = &drupal_static(__FUNCTION__);
	const sicknessReasonsCacheKey = 'civihr_absence_sickness_reasons';
	
    if (!isset($sicknessReasons)) {
	    $cachedsicknessReasons = cache_get(sicknessReasonsCacheKey);
        if ($cachedsicknessReasons) {
            $sicknessReasons = $cachedsicknessReasons->data;
        }
        else {
		    civicrm_initialize();
            $sicknessReasons = ;//Replace with API call and check for count > 0 otherwise default to empty array;
			
            // Cache the absence sickness reasons for 5 minutes
			const secondsInFiveMinutes = 300; 
            cache_set(sicknessReasonsCacheKey, $sicknessReasons, 'cache', time() + secondsInFiveMinutes);
        }
    }
	
	$sicknessReasonLabel = '';
	if (!empty($sicknessReasons[$sicknessReasonValue]) {
	    $sicknessReasonLabel = $sicknessReasons[$sicknessReasonValue];
	}

    return $sicknessReasonLabel;
}
```